js/utils.js

```diff
@@ -1511,6 +1511,33 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     });
   }
 
+  function setFilterValues(options) {
+    options = options || {};
+
+    if (!options.config) {
+      return Promise.resolve();
+    }
+
+    return new Promise(function(resolve) {
+      Fliplet.User.getCachedSession()
+        .then(function(session) {
+          if (options.config.filterOptions.length) {
+            return Promise.all(_.map(options.config.filterOptions, function(item) {
+              switch (item.valueType) {
+                case 'user-profile-data':
+                  item.value = session.entries ? session.entries.dataSource.data[item.fieldValue] : '';
```

Comment:
Your previous version of the code checked for all other passport too. I don't understand why that was removed. I provided some [code sample](https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/464#discussion_r547386465) for structural purpose. It looks like there hasn't been any thought in including the good work that was previously done.

Reply:
@tonytlwu 
This is not true, your example code was very useful, but why did I think that the version was incomplete after deleting comments. i went back to the previous version of the code, and got a little confused about the implementation sorry